### PR TITLE
Properly populate ARM return object

### PR DESCRIPTION
**Pull request by dannysongg**
Ensure ARM requests to OneDeploy are returned with properly populated DeployResult object and Deployment ID.

### CODE REVIEW

**Comment by dannysongg** [comment]
@jvano @dfuenzalida @gregorygoldshteyn @caovicto

**Review by jvano — APPROVED** [review]
Looks good, :shipit:

File: Kudu.Core/Deployment/FetchDeploymentManager.cs [hunk 1]
```diff
@@ -329,6 +330,11 @@ private bool ShouldDeploy(IRepository repository, DeploymentInfoBase deploymentI
             return true;
         }
 
+        public DeployResult GetDeployResult()
+        {
+            return _deploymentManager.GetResults().First();
```
[the remark this answers is not in the document]
Comment (dannysongg) [reply Kudu.Core/Deployment/FetchDeploymentManager.cs:335]: `GetResults()` returns all previous deployments. This code path should never get hit before a deployment happens and failed deployment will still return an object so `GetResults()` should never return an empty collection.

**Review by dfuenzalida — APPROVED** [review]

File: Kudu.Services/Deployment/PushDeploymentController.cs [hunk 2]
```diff
@@ -480,6 +481,13 @@ private async Task<HttpResponseMessage> PushDeployAsync(ArtifactDeploymentInfo d
 
             var response = Request.CreateResponse();
 
+            if (ArmUtils.IsArmRequest(Request))
+            {
+                DeployResult deployResult = _deploymentManager.GetDeployResult();
+                deployResult.Id = deploymentInfo.DeploymentTrackingId;
+                response = Request.CreateResponse(HttpStatusCode.Accepted, ArmUtils.AddEnvelopeOnArmRequest(deployResult, Request));
+            }
```
Comment (suwatch) [inline Kudu.Services/Deployment/PushDeploymentController.cs:489]: This is a breaking change, correct?   Please put it under ScmHost... knob to allow safe rollout.

**Comment by dannysongg** [comment]
@suwatch this is ready to be merged

**Pull request merged**